app/controllers/admin_comment_controller.rb

```diff
@@ -81,7 +78,7 @@ def set_comment
     @comment = Comment.find(params[:id])
   end
 
-  def comment_hidden?(old_visibility, old_body)
-    !@comment.visible && old_visibility && old_body == @comment.body
+  def comment_hidden?
+    @comment.previous_changes[:visible] && !@comment.previous_changes[:body]
```

Comment:
We now get a `hide_comment` event when you change `visible` from `false` to `true`. Given the next commit, I think we should preserve the existing behaviour here. Could either re-add `!@comment.visible` or change to:

```ruby
@comment.previous_changes[:visible] == [true, false] &&
  !@comment.previous_changes[:body]
```